src/Router.php

```diff
@@ -119,44 +119,42 @@ protected function matchDirect(string $httpMethod, string $urlToMatch): ?Route
         return $route;
     }
 
-    protected function matchParametered(string $httpMethod, string $urlToMatch): ?Route
+    protected function matchParameteredOrCatchAll(string $httpMethod, string $urlToMatch): ?Route
     {
         if (!array_key_exists($httpMethod, $this->routes)) {
             return null;
         }
 
         foreach ($this->routes[$httpMethod] as $routeUrl => $route) {
-            if (!str_contains($routeUrl, '{') || !$route->supportsHttpMethod($httpMethod)) {
```

Comment:
This check is done to drop out quickly if we're not going to have to go through the parameter logic. Imagine 800 urls, 400 of which are POST and of the remaining 400 GETs there's only 5 with parameters.

Before: only check the 5 parametered GET requests
Now: go through all 800

Please put this check back or something that has the same effect.

Reply:
I'll look into this